src/lib/device/device_manager.cpp

```diff
@@ -204,11 +206,11 @@ shared< VirtualDev > DeviceManager::create_vdev(vdev_parameters&& vparam) {
         pdevs.erase(pdevs.begin() + 1, pdevs.end()); // TODO: Pick random one
     }
 
-    // Adjust the maximum number chunks requested before round up vdev size.
-    uint32_t max_num_chunks = 0;
-    for (const auto& d : m_dev_infos) {
-        max_num_chunks += hs_super_blk::max_chunks_in_pdev(d);
-    }
+    // Based on the min chunk size, we calculate the max number of chunks that can be created in each target pdev
+    uint32_t min_chunk_size = hs_super_blk::min_chunk_size(vparam.dev_type);
+    // FIXME: it is possible that each vdev is less than max_num_chunks, but total is more than MAX_CHUNKS_IN_SYSTEM.
+    // uint32 convert is safe as it only overflow when vdev size > 64PB with 16MB min_chunk_size.
```

Comment:
should we add an assert to make sure vdev size is smaller than or eaqual to 64 PB?

Reply:
I dont think we can have a vdev > 64PB in any near future.